File: leetcode/0338. Counting Bits/Solution.swift
// https://leetcode.com/problems/counting-bits/

class Solution {
	func countBits(_ num: Int) -> [Int] {
	  var r = [0]
	  if num < 1 { return r }

	  for i in 1...num {
	    if i.isMultiple(of: 2) {
	      r.append(r[i/2])
	    } else {
	      r.append(r[i-1] + 1)
	    }
	  }

	  return r
	}
}

File: leetcode/0338. Counting Bits/Solution.py
# https://leetcode.com/problems/counting-bits/

from typing import List

class Solution:
    def countBits(self, num: int) -> List[int]:
        r = [0]
        if num < 1:
            return r

        for i in range(1, num + 1):
            if not i % 2:
                r.append(r[i // 2])
            else:
                r.append(r[i - 1] + 1)

        return r

s = Solution()
assert s.countBits(2) == [0,1,1]
assert s.countBits(5) == [0,1,1,2,1,2]

# [0-1], [2-3], [4-7], [8-15]...
